$OpenBSD: patch-magick_property_c,v 1.1 2012/04/27 08:09:56 jasper Exp $

Chunk 1: Security fix for CVE-2012-1610, Potential EXIF Interger Overflow
Chunk 2: Security fix for CVE-2012-0259, JPEG EXIF tag crash.

--- magick/property.c.orig	Fri Apr 27 09:39:10 2012
+++ magick/property.c	Fri Apr 27 09:46:04 2012
@@ -1287,6 +1287,8 @@ static MagickBooleanType GetEXIFProperty(const Image *
         break;
       components=(ssize_t) ((int) ReadPropertyLong(endian,q+4));
       number_bytes=(size_t) components*tag_bytes[format];
+      if (number_bytes < components)
+        break; /* prevent overflow */
       if (number_bytes <= 4)
         p=q+8;
       else
@@ -1308,6 +1310,8 @@ static MagickBooleanType GetEXIFProperty(const Image *
             buffer[MaxTextExtent],
             *value;
 
+	  value=(char *) NULL;
+	  *buffer='\0';
           switch (format)
           {
             case EXIF_FMT_BYTE:
